### ISSUE TITLE
The question about script code?

### ISSUE THREAD

**Issue by qingchengleiqing** [ultralytics/ultralytics#14984]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

The dataset has been divided into training set, validation set, and test set. During training, the training set was used. For validation, the trained weights from 'best.pt' were used. I have a question: should the validation be done using the validation set or the test set? If validation uses the validation set, then where should the test set be used? Could you please provide Python script code for training, validation, prediction, and inference?

**Comment by pderrenger** [comment]
@qingchengleiqing for model evaluation, the validation set should be used during training to tune hyperparameters and prevent overfitting. The test set is reserved for final evaluation to assess the model's performance on unseen data. This ensures an unbiased evaluation of the model's generalization capability. If you need further guidance on using the validation and test sets, please refer to our detailed documentation. If you require specific code examples, feel free to ask, and I'll provide minimal examples to get you started.

**Comment by qingchengleiqing** [comment]
Is that correct?
![1](https://github.com/user-attachments/assets/12b5d637-15b4-420e-9915-67a590eeff8a)
train.py
import warnings
warnings.filterwarnings('ignore')
from ultralytics import YOLO

# 训练参数官方详解链接：https://docs.ultralytics.com/modes/train/#resuming-interrupted-trainings:~:text=a%20training%20run.-,Train%20Settings,-The%20training%20settings

if __name__ == '__main__':
    model = YOLO('ultralytics/cfg/models/v8/yolov8n.yaml')
    # model.load('yolov8n.pt') # loading pretrain weights
    model.train(data='dataset/vehicle/data.yaml',
                cache=False,
                imgsz=640,
                epochs=300,
                batch=32,
                close_mosaic=0,
                workers=4,
                # device='0',
                optimizer='SGD', # using SGD
                # patience=0, # close earlystop
                # resume=True, # 断点续训,YOLO初始化时选择last.pt
                # amp=False, # close amp
                # fraction=0.2,
                project='runs/train',
                name='exp',
                )
val.py
import warnings
warnings.filterwarnings('ignore')
from ultralytics import YOLO

# 验证参数官方详解链接：https://docs.ultralytics.com/modes/val/#usage-examples:~:text=of%20each%20category-,Arguments%20for%20YOLO%20Model%20Validation,-When%20validating%20YOLO

if __name__ == '__main__':
    model = YOLO('runs/train/exp/weights/best.pt')
    model.val(data='dataset/vehicle/data.yaml',
              split='test',
              imgsz=640,
              batch=16,
              # iou=0.7,
              # rect=False,
              # save_json=True, # if you need to cal coco metrice
              project='runs/val',
              name='exp',
              )

**Comment by Y-T-G** [comment]
`test` is not used. It's upto you. You can use it after training by running `model.val(..., split="test")`
